tests/test_picking_batch.py

```diff
@@ -305,3 +307,219 @@ def test14_is_valid_location_dest_failure_unknown_location(self):
+        found
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.update_picking(force_validate=True,
+                               location_dest_id=self.test_output_location_01.id)  # noqa
+        reason = 'missing item'
+
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=999,
+                                  reason=reason,
+                                  picking_type_id=None)
+            self.assertEqual(err.exception.name,
+                             'Cannot find the operation')
+
+    def test18_unpickable_item_wrong_batch(self):
+        """
+        Tests that a ValidationError is raised if the move_line_id is not on
+        the Batch that we requested.
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        # Create a quant and picking for a different package
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_two.id)
+        different_picking = self.create_picking(
+            self.picking_type_pick,
+            products_info=self.pack_4apples_info,
+            confirm=True,
+            assign=True)
+        move_line_id = different_picking.move_line_ids[0].id
+        reason = 'missing item'
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=move_line_id,
+                                  reason=reason,
+                                  picking_type_id=None)
+            self.assertEqual(err.exception.name,
+                             'Move line is not part of the batch')
+
+    def test19_unpickable_item_invalid_state_cancel(self):
+        """
+        Tests that a ValidationError is raised if the picking is on a state
+        of cancel
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        # Not ideal but it allows the test to pass.  If we did:
+        # picking.action_cancel() it would delete the move_lines which would
+        # cause this test to fail incorrectly.
+        picking.state = 'cancel'
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=move_line_id,
+                                  reason=reason,
+                                  picking_type_id=None)
+            self.assertEqual(err.exception.name,
+                             'Cannot mark a move line as unpickable '
+                             'when it is part of a completed Picking')
+
+    def test20_unpickable_item_invalid_state_done(self):
+        """
+        Tests that a ValidationError is raised if the picking is on a state of
+        done
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.state = 'done'
```

Comment:
In this case we want to use update_picking with force_validate and location_dest_id, like in test11, instead of manually setting the state of the picking.